src/azure-cli/azure/cli/command_modules/vm/custom.py

```diff
@@ -887,6 +888,43 @@ def create_vm(cmd, vm_name, resource_group_name, image=None, size='Standard_DS1_
     return vm
 
 
+def auto_shutdown_vm(cmd, resource_group_name, vm_name, off=None, email=None, webhook=None, time=None,
+                     location=None):
+    from msrestazure.tools import resource_id
+    from azure.mgmt.devtestlabs.models import Schedule
+    client = _dev_test_labs_client_factory(cmd.cli_ctx)
+    name = 'shutdown-computevm-' + vm_name
+    vm_id = resource_id(subscription=client.config.subscription_id, resource_group=resource_group_name,
+                        namespace='Microsoft.Compute', type='virtualMachines', name=vm_name)
+    if off:
+        if email is not None or webhook is not None or time is not None:
+            # I don't want to disturb users. So I warn instead of raising an error.
+            logger.warning('If --off, other parameters will be ignored.')
+        return client.global_schedules.delete(resource_group_name, name)
+
+    if time is None:
+        raise CLIError('usage error: --time is a required parameter')
+    if email is not None and webhook is None:
```

Comment:
If a user gives `""` for `email`, `email is not None` evaluates to `True`.

```suggestion
    if email and not webhook:
```

Reply:
"" is different from None. "" means user intends to set "" explicitly while None means user doesn't set it and leave it to default. I respect user's intention.

Reply:
Empty `email` or `webhook` doesn't make too much sense to me.